src/tool_operate.c

```diff
@@ -2163,6 +2166,18 @@ static CURLcode parallel_transfers(struct GlobalConfig *global,
   }
 
   while(!mcode && (still_running || more_transfers)) {
+    /* If stopping prematurely (eg due to a --fail-early condition) then signal
+       that any transfers in the multi should abort (via progress callback). */
+    if(wrapitup) {
+      struct per_transfer *per;
+      if(!still_running)
+        break;
+      for(per = transfers; per; per = per->next) {
+        if(per->added)
+          per->abort = TRUE;
+      }
+    }
```

Comment:
Do we really need this loop here to run every lap? Isn't it enough to iterate over all the transfers once when `wrapitup` is set to true?

Reply:
> Do we really need this loop here to run every lap? Isn't it enough to iterate over all the transfers once when `wrapitup` is set to true?

> Also: when wrapitup is set TRUE, shouldn't this then work as a condition to not set checkmore = TRUE, to prevent more transfers (if available) to get added to the queue?

I wrote it to skip the checkmore action by continuing the loop. We don't need to tend to checkmore or more_transfers since the wrapitup action at the beginning of the loop overrides it. Setting each transfer's abort flag only once should do it since no other transfers are added (ie the checkmore job is skipped). I've addressed this in 7fa96e3.